experimental/uefi/kernel/src/interrupts.rs

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::i8042;
+use core::ops::Deref;
+use lazy_static::lazy_static;
+use log::error;
+use x86_64::structures::idt::{InterruptDescriptorTable, InterruptStackFrame};
+
+lazy_static! {
+    static ref IDT: InterruptDescriptorTable = {
+        let mut idt = InterruptDescriptorTable::new();
+        idt.double_fault.set_handler_fn(double_fault_handler);
+        idt
+    };
+}
+
+extern "x86-interrupt" fn double_fault_handler(
```

Comment:
What is "x86-interrupt"? Is this a type of linker that the Rust compiler already knows about or is it defined somewhere? Could you add a clarifying comment?

Reply:
No, it's not about linker -- it changes the calling convention of the function and what code the compiler emits. The particular calling convention is required by the x86-64 ABI.

You can read more about the `x86-interrupt` in particular here: https://github.com/rust-lang/rust/issues/40180

In short, you don't really expect to see that used anywhere else except for interrupt handlers.

Reply:
I think a comment with a link to that issue would help us understand what is going on later when we look at the code.